components/serverless/internal/controllers/serverless/utils.go

```diff
@@ -186,22 +186,48 @@ func mapsEqual(existing, expected map[string]string) bool {
 	return true
 }
 
+func mapsContains(existing, contains map[string]string) bool {
+	if len(existing) < len(contains) {
+		return false
+	}
+
+	for key, value := range contains {
+		if v, ok := existing[key]; !ok || v != value {
+			return false
+		}
+	}
+
+	return true
+}
+
+func mergeMapWithNewValues(existing, newValues map[string]string) {
+	for key, value := range newValues {
+		existing[key] = value
+	}
+}
+
 // TODO refactor to make this code more readable
 func equalDeployments(existing appsv1.Deployment, expected appsv1.Deployment) bool {
-	return len(existing.Spec.Template.Spec.Containers) == 1 &&
-		len(existing.Spec.Template.Spec.Containers) == len(expected.Spec.Template.Spec.Containers) &&
-		existing.Spec.Template.Spec.Containers[0].Image == expected.Spec.Template.Spec.Containers[0].Image &&
-		envsEqual(existing.Spec.Template.Spec.Containers[0].Env, expected.Spec.Template.Spec.Containers[0].Env) &&
-		mapsEqual(existing.GetLabels(), expected.GetLabels()) &&
-		mapsEqual(existing.Spec.Template.GetLabels(), expected.Spec.Template.GetLabels()) &&
-		equalResources(existing.Spec.Template.Spec.Containers[0].Resources, expected.Spec.Template.Spec.Containers[0].Resources) &&
-		equalInt32Pointer(existing.Spec.Replicas, expected.Spec.Replicas) &&
-		equalSecretMounts(existing.Spec.Template.Spec, expected.Spec.Template.Spec) &&
-		mapsEqual(existing.Spec.Template.GetAnnotations(), expected.Spec.Template.GetAnnotations())
+	result := true
```

Comment:
IMO don't touch this or do refactor by excluding parts of code to functions with good names ;)

Reply:
later ;)